pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -158,6 +155,32 @@ func warnDeprecatedAPIs(spec *v1.VirtualMachineInstanceSpec, config *virtconfig.
 	return warnings
 }
 
+func ValidateVirtualMachineInstancePerArch(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	arch := spec.Architecture
+
+	switch arch {
+	case "amd64":
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceAmd64Setting(field, spec)...)
+	case "s390x":
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceS390XSetting(field, spec)...)
+	case "arm64":
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceArm64Setting(field, spec)...)
+	case "ppc64le":
+		// no validation needed for ppc64le
+	case "":
+		// validation is not needed using default architecture
```

Comment:
I would drop this case, since in the validation stage the `spec.Architecture` should have been set by the mutating webhook. So this case should never happen.

Reply:
nice catch!
in this case, let's just error out if the arch is empty.